src/js/utils/html-utils.js

```diff
@@ -97,7 +79,7 @@
          *
          * @param {string} formatString The string to be interpolated.
          * @param {Object} parameters An optional set of parameters to the template.
-         * @returns {HtmlSnippet} The resulting safely escaped HTML snippet.
+         * @returns {string} The resulting safely escaped HTML snippet.
```

Comment:
Do you want to define somewhere that an "HTML snippet" is a "string flagged as an HTML snippet"?  Or, would it be better to consistently use the latter through all comments?
For example:
The resulting safely escaped string flagged as an HTML snippet.